### ISSUE TITLE
SC2236 proposes using '-n' but it doesn't replace '! -z'  in all cases

### ISSUE THREAD

**Issue by hunter86bg** [koalaman/shellcheck#2475]
#### For bugs
- Rule Id (if any, e.g. SC1000):  SC2236
- My shellcheck version (`shellcheck --version` or "online"):  VS Code (v0.18.10)
- [X] The rule's wiki page does not already cover this (e.g. https://shellcheck.net/wiki/SC2086)
- [X] I tried on https://www.shellcheck.net/ and verified that this is still a problem on the latest commit

#### For new checks and feature suggestions
- [ ] https://www.shellcheck.net/ (i.e. the latest commit) currently gives no useful warnings about this
- [X] I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything related

#### Here's a snippet or screenshot that shows the problem:

```sh
#!/usr/bin/bash

var=1

if [ ! -z ${var+x} ]; then
        echo "var is set to '$var'"
fi
```

#### Here's what shellcheck currently says:
Use -n instead of ! -z

#### Here's what I wanted or expected to see:
The recommendation is wrong in some cases like this one:

```sh
$ cat ./test.sh

#!/usr/bin/bash

#var=1

if [ -n ${var+x} ]; then
        echo "var is set to '$var'"
else
        echo "var is undefined"
fi

$ ./test.sh
var is set to ''

$ cat test.sh
#!/usr/bin/bash

#var=1

if [ ! -z ${var+x} ]; then
        echo "var is set to '$var'"
else
        echo "var is undefined"
fi

$ ./test.sh
var is undefined
```

**Comment by brother** [comment]
This example isn't the best in the world as following the advise to switch `! -z` into `-n` will then advise you to quote the variable (or switch to double brackets testing).

**Comment by hunter86bg** [comment]
Well, it's not recommending quiting the var or using double brackets

**Comment by hunter86bg** [comment]
Closing and opening the VS code fixed the situation as I got warned that I need to quote or use double brackets.

**Comment by brother** [comment]
Not sure where you are looking but it actually 100% do recommend that. 
Great to hear that you got it working.

![image](https://user-images.githubusercontent.com/243351/160157320-4946aec2-e025-4e67-93d8-5a38a9c8f27e.png)
